tests/rados/test_scrub_parameters.py

```diff
@@ -0,0 +1,252 @@
+"""
+This method is used to verify the dump scrub.
+"""
+
+import datetime
+import random
+import time
+import traceback
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from ceph.rados.rados_scrub import RadosScrubber
+from tests.rados.monitor_configurations import MonConfigMethods
+from tests.rados.stretch_cluster import wait_for_clean_pg_sets
+from utility.log import Log
+from utility.utils import method_should_succeed
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    1. Test to the forced flag during the scrubbing and deep-scrubbing
+    2. Test the scheduled time <=  osd_scrub_min_interval+
+        (osd_scrub_min_interval * osd_scrub_interval_randomized_ratio)
+    Returns:
+        1 -> Fail, 0 -> Pass
+    """
+
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_object = RadosOrchestrator(node=cephadm)
+    scrub_object = RadosScrubber(node=cephadm)
+    mon_obj = MonConfigMethods(rados_obj=rados_object)
+    pool_type = ["replicated_pool", "ec_pool"]
+    try:
+
+        flag = wait_for_clean_pg_sets(rados_object)
+        if not flag:
+            log.error(
+                "The cluster did not reach active + Clean state after add capacity"
+            )
+            return 1
+        for val in ("scrub", "deep-scrub"):
+            for pool in pool_type:
+                pool_detail = config[pool]
+                pool_name = pool_detail["pool_name"]
+                selected_osd = create_pool_get_osd(rados_object, pool, pool_detail)
+                dump_scrub_before_scrub = scrub_object.get_dump_scrubs(selected_osd)
+                if val == "scrub":
+                    rados_object.run_scrub(osd=selected_osd)
+                else:
+                    rados_object.run_deep_scrub(osd=selected_osd)
+                status, pg_id = chk_forced_flag(scrub_object, selected_osd)
```

Comment:
Is the force flag updated on the PG soon after the scrub/deep-scrub is triggered on the OSD?

Do we need to add some sleep time here?

And if the OSD scrub/deep-scrub did not trigger force flag, how is triggering on PG a workaround?

Reply:
The method name "chk_forced_flag" is changed to "is_forced_flag_set_true", and time is added at the called method. 